eras/shelley/impl/src/Cardano/Ledger/Shelley/LedgerState.hs

```diff
@@ -1226,17 +1259,73 @@ incrementalStakeDistr ::
   SnapShot (Crypto era)
 incrementalStakeDistr incstake ds ps =
   SnapShot
-    (Stake $ VMap.fromMap (compactCoinOrError <$> eval (dom activeDelegs ◁ stake1)))
-    (VMap.fromMap delegs)
+    (Stake $ VMap.fromMap (compactCoinOrError <$> step2))
+    (VMap.fromMap (UM.unUnify delegs))
     (VMap.fromMap poolParams)
   where
-    DState rewards' delegs bimap _ _ _ = ds
+    UnifiedMap tripmap ptrmap = _unified ds
     PState poolParams _ _ = ps
-    stake0, stake1 :: Map (Credential 'Staking (Crypto era)) Coin
-    stake0 = resolveIncrementalPtrs @era (forwards bimap) incstake
-    stake1 = Map.unionWith (<>) stake0 rewards'
-    activeDelegs :: Map (Credential 'Staking (Crypto era)) (KeyHash 'StakePool (Crypto era))
-    activeDelegs = eval ((dom rewards' ◁ delegs) ▷ dom poolParams)
+    delegs = delegations ds
+    step1 = resolveActiveIncrementalPtrs (activeP tripmap) ptrmap incstake
+    step2 = aggregateActiveStake tripmap step1
+
+-- | A credential is active, only if the the third part of the triple is (SJust _)
+activeP :: Map (Credential 'Staking crypt0) (Triple crypto) -> Credential 'Staking crypt0 -> Bool
```

Comment:
what does the `P` stand for?

Reply:
Predicate?  Its a Lisp thing. predicates (functions to True and False) often end in P.